client/my-sites/domains/domain-management/transfer/transfer-out/unlocked.jsx

```diff
@@ -62,81 +99,193 @@ class Unlocked extends React.Component {
 						break;
 				}
 				notices.error( errorMessage );
-
-				this.setState( { submitting: false } );
-			} else if ( hasPrivacyProtection ) {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked and ' +
-					'Privacy Protection has been enabled.' ) );
 			} else {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked back.' ) );
+				// Success.
+				this.setStateIfMounted( { sent: false } );
+
+				let successMessage;
+				if ( enablePrivacy && lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer. Your domain is now re-locked and ' +
+						'Privacy Protection has been enabled.' );
+				} else if ( enablePrivacy ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						'Privacy Protection has been re-enabled.' );
+				} else if ( lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						're-locked your domain.' );
+				} else {
+					successMessage = translate( 'We\'ve canceled your domain transfer. ' );
+				}
+
+				notices.success( successMessage );
 			}
 		} );
 	};
 
-	handleResendConfirmationCodeClick = () => {
-		this.setState( { submitting: true } );
+	isDomainAlwaysTransferrable() {
+		const {
+			domainLockingAvailable,
+			hasPrivacyProtection,
+		} = getSelectedDomain( this.props );
+		return ! domainLockingAvailable && ! hasPrivacyProtection;
+	}
+
+	renderCancelButton( domain ) {
+		const { pendingTransfer } = domain;
+
+		if ( ! pendingTransfer && this.isDomainAlwaysTransferrable() ) {
```

Comment:
Maybe I should swap the clauses around.

If it's pending transfer, we always want to show the cancel button, because we can cancel the pending transfer, so `! pendingTransfer`.

Otherwise, cancelling the transfer just means "reinstate privacy and the lock, as appropriate", so for "alwaysTransferrable", that's a null-op, so we should skip the cancel button. Unless there's a pending transfer.

I put the clauses this way around because pending transfer is cheaper, but it's not a meaningful difference, and `this.isDomainAlwaysTransferrable() && ! pendingTransfer` will probably read better - "Skip the cancel button domains that don't need it, unless they're pending"